src/app/services/announcement/announcement.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { Announcement, AnnouncementType } from './announcement';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AnnouncementService {
+  readonly newAnnouncement$: ReplaySubject<Announcement>;
+
+  constructor() {
+    this.newAnnouncement$ = new ReplaySubject<Announcement>(1);
+  }
+
+  announce(aore: Announcement | Error) {
+    if (this.isAnnouncement(aore)) {
```

Comment:
This is my opinion here, but I think lines 16-20 would be more readable if there were written a single line trinary. Feel free to take or leave that suggestion though.